src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,84 @@
+  const [isModalLoading, setIsModalLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    setIsModalLoading(true);
+
+    if (!currentTodo) {
+      return;
+    }
+
+    getUser(currentTodo.userId)
+      .then(data => setUser({ ...data }))
+      .finally(() => setIsModalLoading(false));
+  }, [currentTodo]);
+
+  const handleCloseModal = () => {
+    dispatch(setCurrentTodo(null));
+  };
+
   return (
-    <div className="modal is-active" data-cy="modal">
+    <div
+      className={classNames('modal', { 'is-active': currentTodo?.id !== 0 })}
+      data-cy="modal"
+    >
       <div className="modal-background" />
 
-      <Loader />
+      {isModalLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{currentTodo?.id}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              onClick={handleCloseModal}
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+            />
+          </header>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {currentTodo?.title}
+            </p>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+            <p className="block" data-cy="modal-user">
+              {/* <strong className="has-text-success">Done</strong> */}
+              {currentTodo?.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              {' by '}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
It's a good idea to check if the user data exists before trying to access its properties. This can help avoid potential errors when the user data is undefined.